server/src/inmantals/lsp_types.py

```diff
@@ -36,15 +40,21 @@ class LspModel(BaseModel):
 
     class Config:
         alias_generator = partial(re.sub, r"_([a-z])", lambda match: match.group(1).upper())
-        allow_population_by_field_name = True
+        if SUPPORTS_PYDANTIC_V2:
+            populate_by_name = True
+        else:
+            allow_population_by_field_name = True
```

Comment:
how does this work? I mean where is this used?

Reply:
You can see this in action in these tests: [test_lsp_type_serialization_pydantic_v1](https://github.com/inmanta/vscode-inmanta/blob/b99e8494eeb3a403284503b44a1130777d3ae3ab/server/tests/test_smoke.py#L393)
and [test_lsp_type_serialization_pydantic_v2](https://github.com/inmanta/vscode-inmanta/blob/b99e8494eeb3a403284503b44a1130777d3ae3ab/server/tests/test_smoke.py#L416) 
It's used to allow populating the model with either the field name or its alias (see [pydantic doc](https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.populate_by_name) for details)